script/utils/update-redirects-data.ts

```diff
@@ -60,10 +69,20 @@ const updateRedirectsData = async(currentBook: BookWithOSWebData, newBook: BookW
       countNewRedirections++;
     // remove `else` to enable legitimately removing pages from books
     // only once uuids are guaranteed to be consistent
-    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException) {
+    } else if (!newSection && matchSlug(section.slug) === undefined && !matchException && !allowBookRedirect) {
```

Comment:
Once you get past the initial validation, I'd change these checks to compare the book ids, I think that is more clear. Like in this case this check is only used if the book ids are the same